api/src/main/java/org/openmrs/api/db/hibernate/HibernateUserDAO.java

```diff
@@ -106,6 +106,26 @@ public User getUserByUsername(String username) {
 		return users.get(0);
 	}
 	
+	
+	/**
+	 * @see org.openmrs.api.UserService#getUserByEmail(java.lang.String)
+	 */
+	@Override
+	@SuppressWarnings("unchecked")
+	public User getUserByEmail(String email) {
+		Query query = sessionFactory.getCurrentSession().createQuery(
```

Comment:
Can you please use criteria API since it is cleaner? This would be one line of code